src/solvers/refinement/string_refinement.cpp

```diff
@@ -928,6 +928,70 @@ void string_refinementt::substitute_array_access(exprt &expr) const
 /// to avoid false positives the symbols other than the universal variable
 /// should have been replaced by there valuation in the current model
 /// \par parameters: a string constraint and a solver for non string expressions
+void string_refinementt::add_negation_of_not_contains_constraint_to_solver(
+  const string_not_contains_constraintt &axiom, symbol_exprt &univ_var,
+  exprt val, supert &solver)
+{
+  exprt lbu=axiom.univ_lower_bound();
+  exprt ubu=axiom.univ_upper_bound();
+  if(lbu.id()==ID_constant && ubu.id()==ID_constant)
+  {
+    mp_integer lb_int, ub_int;
+    to_integer(to_constant_expr(lbu), lb_int);
+    to_integer(to_constant_expr(ubu), ub_int);
+    if(ub_int<=lb_int)
+    {
+      debug() << "empty constraint with current model" << eom;
+      solver << false_exprt();
+      return;
+    }
+  }
+
+  exprt lbe=axiom.exists_lower_bound();
+  exprt ube=axiom.exists_upper_bound();
+  if(lbe.id()==ID_constant && ube.id()==ID_constant)
+  {
+    mp_integer lb_int, ub_int;
+    to_integer(to_constant_expr(lbe), lb_int);
+    to_integer(to_constant_expr(ube), ub_int);
+    if(ub_int<=lb_int)
+    {
+      debug() << "empty constraint with current model" << eom;
+      solver << false_exprt();
```

Comment:
since the upper bound is smaller than the lower bound, there should be no possible existential assignment, so the formula should be trivially false and its negation true, so I think `true_exprt` needs to be added there.